src/openrct2/ride/Ride.cpp

```diff
@@ -5656,9 +5656,10 @@ rct_ride_name get_ride_naming(const uint8_t rideType, rct_ride_entry* rideEntry)
     }
 }
 
-bool ride_type_has_flag(int32_t rideType, uint32_t flag)
+bool ride_type_has_flag(int32_t rideType, uint64_t flag)
 {
-    return (RideProperties[rideType].flags & flag) != 0;
+    Guard::Assert(rideType < static_cast<int32_t>(std::size(RideTypeDescriptors)));
+    return (RideTypeDescriptors[rideType].Flags & flag) != 0;
```

Comment:
Guard::Assert will always crash the game, so returning something is pointless. You're thinking of `assert()`, for which your suggestion would be more appropriate.

Reply:
Guard assert crashes even in release? I didn't know that. I thought it was just an enhanced `assert`

Reply:
Hm, on closer inspection of the code, it might not actually crash the game, or not always. So your suggestion might be of use. By the way, you can just `Guard::Assert(false)`, since you moved the check to the if/else.